pyravendb/data/document_conventions.py

```diff
@@ -65,27 +69,46 @@ def _json_default(o):
         else:
             raise TypeError(repr(o) + " is not JSON serializable (Try add a json default method to store convention)")
 
+    @property
+    def default_collection_names(self) -> dict:
+        return self._default_collection_names
+
+    @default_collection_names.setter
+    def default_collection_names(self, names):
+        if self._frozen:
+            raise InvalidOperationException(
+                "Conventions has frozen after store.initialize() and no changes can be applied to them")
+        self._default_collection_names.update(names)
+
+    def get_collection_name(self, entity: Type) -> str:
+        if entity in self._default_collection_names:
+            return self._default_collection_names[entity]
+        else:
+            return DocumentConventions.default_transform_plural(entity.__name__)
+
     @staticmethod
     def default_transform_plural(name):
         return inflector.plural(name)
 
-    @staticmethod
-    def default_transform_type_tag_name(name):
+    def default_transform_type_tag_name(self, entity: Type):
+        if entity.__class__ in self._default_collection_names:
+            name = self._default_collection_names[entity.__class__]
+        else:
+            name = DocumentConventions.default_transform_plural(entity.__class__.__name__)
         count = sum(1 for c in name if c.isupper())
         if count <= 1:
-            return DocumentConventions.default_transform_plural(name.lower())
-        return DocumentConventions.default_transform_plural(name)
+            return name.lower()
```

Comment:
This is a mistake, default_transform_plural uses ` inflector.plural(name)`
`DocumentConventions.default_transform_plural(name.lower())` return the name from the package inflect,
here we just check if what `default_collection_names` or `default_transform_plural` returns and messed that up by call lower on the string.

Reply:
Hi @IdanHaim 

inflector.plural returns the same result regardless of whether we pass the original or lower string as a parameter

```
>>> import inflect
>>> inflector = inflect.engine()
>>> inflector.plural('order')
'orders'
>>> inflector.plural('Order')
'Orders'
>>> inflector.plural('OrderItem')
'OrderItems'
>>> inflector.plural('orderitem')
'orderitems'
>>>
```

The default_transform_type_tag_name is intended to generate the object's key, and unless I get it wrong, Raven uses the following rules:

a) If it is a simple name (like Company), the key will be the plural name in lowercase: "companies/XXX"

b) If the name is composed (like Company Owner), the key will be the name with capital initials and the last name pluralized: "CompanyOwners"

What I do then is, if the class doesn't have a custom name, I first pluralize it and then adjust the result to the RavenDB standard. The snippet you referenced will only be called if the name is a single word.

You can a sample here:

```
    def test_transform_type_name(self):
        dc = DocumentConventions()
        assert "companies" == dc.default_transform_type_tag_name(Company("Test"))
        assert "CompanyOwners" == dc.default_transform_type_tag_name(CompanyOwner("Test Test"))
        assert "OrderItems" == dc.default_transform_type_tag_name(OrderItem(1, 1))
        assert "categories" == dc.default_transform_type_tag_name(Category("category"))
        assert "DemoStats" == dc.default_transform_type_tag_name(DemoStat())
```
https://gist.github.com/cassioeskelsen/80e0a31516b6b32ff7a8963a34b2d13d

Reply:
I understand you are correct about the plural, please check the behaviour for C# client custom collection names,
do we still change the string to lower if the user has a custom collection name?
if the answer is yes you can resolve this if not the if statement needs to go under the else and only for default.

Reply:
@IdanHaim yes, in my job we have a C# program with this:

```csharp

                    FindCollectionName = type =>
                    {
                        if (type == typeof(Operacao))
                            return "Operacoes";
                        if (type == typeof(CargaTabelas))
                            return "CargasTabelas";
                        (....)
                        return DocumentConventions.DefaultGetCollectionName(type);
                    }                
```

and on Raven database:
Collection: Operacoes,  keys: operacoes/xxxx        (only one word)
Collection: CargasTabelas, keys: CargasTabelas/xxxx   (two words)

PS: default_transform_type_tag_name don't change collection name, only key prefix.

Reply:
The case rules for RavenDB says:

* Starts with a capital letter (only one) - lowercase. User => users/...
* Multiple capital words - as as "FooBar" => FooBars/...